extended/src/main/java/apoc/cypher/CypherExtended.java

```diff
@@ -155,39 +157,77 @@ private <T> BlockingQueue<T> runInSeparateThreadAndSendTombstone(int queueCapaci
         return queue;
     }
 
-    private void runDataStatementsInTx(Scanner scanner, BlockingQueue<RowResult> queue, Map<String, Object> params, boolean addStatistics, long timeout) {
+    private void runDataStatementsInTx(Scanner scanner, BlockingQueue<RowResult> queue, Map<String, Object> params, boolean addStatistics, long timeout, boolean reportError, String fileName) {
         while (scanner.hasNext()) {
             String stmt = removeShellControlCommands(scanner.next());
             if (stmt.trim().isEmpty()) continue;
-            if (!isSchemaOperation(stmt)) {
+            boolean schemaOperation;
+            try {
+                schemaOperation = isSchemaOperation(stmt);
+            } catch (Exception e) {
+                getError(queue, reportError, e, fileName);
+                return;
+            }
+
+            if (!schemaOperation) {
                 if (isPeriodicOperation(stmt)) {
-                    Util.inThread(pools , () -> db.executeTransactionally(stmt, params, result -> consumeResult(result, queue, addStatistics, timeout)));
+                    Util.inThread(pools , () -> {
+                        try {
+                            return db.executeTransactionally(stmt, params, result -> consumeResult(result, queue, addStatistics, timeout));
+                        } catch (Exception e) {
+                            return getError(queue, reportError, e, fileName);
+                        }
+                    });
                 }
                 else {
                     Util.inTx(db, pools, threadTx -> {
                         try (Result result = threadTx.execute(stmt, params)) {
                             return consumeResult(result, queue, addStatistics, timeout);
+                        } catch (Exception e) {
+                            return getError(queue, reportError, e, fileName);
                         }
                     });
                 }
             }
         }
     }
 
+    private Object getError(BlockingQueue<RowResult> queue, boolean reportError, Exception e, String fileName) {
+        if (reportError) {
```

Comment:
I would reverse the condition check:

```suggestion
        if (!reportError) {
```